rabbus.go

```diff
@@ -438,6 +438,15 @@ func (r *Rabbus) handleAmqpClose(err error) {
 		r.mu.Lock()
 		r.Amqp = aw
 		r.mu.Unlock()
+
+		if err := r.WithQos(
+			r.config.qos.prefetchCount,
+			r.config.qos.prefetchSize,
+			r.config.qos.global,
+		); err != nil {
+			continue
```

Comment:
I think we should close the previous created connection in case of WithQos error, so ensure to not leak any connections, wdyt?

Reply:
Sounds reasonable, even though `r.Amqp` will get replaces with a new connection in the next second, and the old connection will get collected gc, if I understand correctly. Anyway, I'll add connection close just to be safe.